pkg/controller/job/job_controller_test.go

```diff
@@ -6505,8 +6778,8 @@ func TestJobBackoffOnRestartPolicyNever(t *testing.T) {
 				t.Errorf("unexpected number of failed pods. Expected %d, saw %d\n", tc.expectedFailed, actual.Status.Failed)
 			}
 			// validate conditions
-			if tc.expectedCondition != nil && !getCondition(actual, *tc.expectedCondition, v1.ConditionTrue, tc.expectedConditionReason) {
-				t.Errorf("expected completion condition. Got %#v", actual.Status.Conditions)
+			if diff := cmp.Diff(tc.expectedConditions, actual.Status.Conditions, cmpopts.IgnoreFields(batch.JobCondition{}, "LastProbeTime", "LastTransitionTime")); diff != "" {
+				t.Errorf("Unexpected conditions (-want,+got):\n%s", diff)
```

Comment:
nit: should we capitalise error messages? I see mostly it starts with lowercase in the file, but it is a mix-and-match of both

Reply:
hm, maybe yes: https://staticcheck.io/docs/checks#ST1005, but I think it does not matter that much, especially for tests, I can adjust touched places.

Reply:
done